### PR TITLE
fix: storytelling widget does not get layers' title

### PR DESCRIPTION
**Pull request by airslice**
# Overview

This bug comes from the change to `layers` state.

In the past the `layers` state will always been set with `new LayerStore()` but now it will be re-used therefore the state will not got changed anymore.

The widget `stories` uses an `useMemo` with `findLayerById` and `storiesData` as dependencies. The `findLayerById` is a method refers to `LayerStore` so it will also not been changed now.

## What I've done

 I added `reearth?.layers?.layers` as `stories`'s dependency so it can been updated now.

## How I tested

- Reload page with a story telling wight;
- Add layer to story telling;

### CODE REVIEW

**Comment by lavalse** [comment]
I have tested. Seems work fine

File: src/components/molecules/Visualizer/Widget/Storytelling/hooks.ts [hunk 1]
```diff
@@ -95,7 +95,8 @@ export default function ({
       ...story,
       title: story.title || (story.layer && findLayerById?.(story.layer)?.title) || "",
     }));
-  }, [findLayerById, storiesData]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [findLayerById, storiesData, reearth?.layers?.layers]);
```
[the remark this answers is not in the document]
Comment (airslice) [reply src/components/molecules/Visualizer/Widget/Storytelling/hooks.ts:99]: Not sure about the backwards but from my understanding i think the update will not trigger more than before bc it was goint to update everytime `layers` got changed, right?

[the remark this answers is not in the document]
Comment (KaWaite) [reply src/components/molecules/Visualizer/Widget/Storytelling/hooks.ts:99]: I just mean the reearth api, in my mind, would be getting its data from the same source as storiesData would. More or less. 
Looking at Storytelling's index.tsx file, I see that `storiesData` is just being assigned a value from widget.property.stories. Can you memoize it and see if it then renders on update instead?

[the remark this answers is not in the document]
Comment (airslice) [reply src/components/molecules/Visualizer/Widget/Storytelling/hooks.ts:99]: The layers data and widgets data comes from different gql as source. The widgets data got memoized and should not update much i think. Inside the widges data it only contains layers' ids and the actual layer data like the title need to be got from layer data. When initialized the widget try to get the layer data before layer data been set so it leads to empty and bc the widgets state not changing later and there's no change for it to get layer data again.

**Review by KaWaite — APPROVED** [review]

**Review by rot1024 — APPROVED** [review]

**Pull request merged**